contracts/interchain-token-service/src/testutils.rs

```diff
@@ -1,2 +1,160 @@
+use soroban_sdk::token::StellarAssetClient;
+use soroban_sdk::{Address, BytesN, Env, IntoVal, String, Vec};
+use soroban_token_sdk::metadata::TokenMetadata;
+use stellar_axelar_gas_service::{AxelarGasService, AxelarGasServiceClient};
+use stellar_axelar_gateway::testutils::{
+    generate_proof, get_approve_hash, setup_gateway, TestSignerSet,
+};
+use stellar_axelar_gateway::types::Message;
+use stellar_axelar_gateway::AxelarGatewayClient;
+use stellar_axelar_std::types::Token;
+
+use crate::{InterchainTokenService, InterchainTokenServiceClient};
+
+#[cfg(any(test, feature = "testutils"))]
 pub const INTERCHAIN_TOKEN_WASM_HASH: &[u8] = include_bytes!("./testdata/interchain_token.wasm");
+
+#[cfg(any(test, feature = "testutils"))]
+pub fn setup_gas_service<'a>(env: &Env) -> AxelarGasServiceClient<'a> {
+    let owner: Address = Address::generate(env);
+    let gas_collector: Address = Address::generate(env);
+    let gas_service_id = env.register(AxelarGasService, (&owner, &gas_collector));
+    let gas_service_client = AxelarGasServiceClient::new(env, &gas_service_id);
+
+    gas_service_client
+}
+
+#[cfg(any(test, feature = "testutils"))]
+pub fn setup_its<'a>(
+    env: &Env,
+    gateway: &AxelarGatewayClient,
+    gas_service: &AxelarGasServiceClient,
+) -> InterchainTokenServiceClient<'a> {
+    let owner = Address::generate(env);
+    let operator = Address::generate(env);
+    let its_hub_address = String::from_str(env, "its_hub_address");
+    let chain_name = String::from_str(env, "chain_name");
+
+    // Note: On changes to `interchain-token` crate, recompile it via `stellar contract build && ./optimize.sh`
+    // and copy the built `target/wasm32-unknown-unknown/release/interchain_token.optimized.wasm` to ../testdata.
+    let interchain_token_wasm_hash = env
+        .deployer()
+        .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_HASH);
+
+    let native_token_address = env.register_stellar_asset_contract_v2(owner.clone());
+
+    let contract_id = env.register(
+        InterchainTokenService,
+        (
+            &owner,
+            &operator,
+            &gateway.address,
+            &gas_service.address,
+            its_hub_address,
+            chain_name,
+            native_token_address.address(),
+            interchain_token_wasm_hash,
+        ),
+    );
+
+    InterchainTokenServiceClient::new(env, &contract_id)
+}
+
+#[cfg(any(test, feature = "testutils"))]
+pub fn setup_env<'a>() -> (
+    Env,
+    InterchainTokenServiceClient<'a>,
+    AxelarGatewayClient<'a>,
+    AxelarGasServiceClient<'a>,
+    TestSignerSet,
+) {
+    let env = Env::default();
+
+    let (signers, gateway_client) = setup_gateway(&env, 0, 5);
+    let gas_service_client: AxelarGasServiceClient<'_> = setup_gas_service(&env);
+
+    let client = setup_its(&env, &gateway_client, &gas_service_client);
+
+    (env, client, gateway_client, gas_service_client, signers)
+}
+
+#[cfg(any(test, feature = "testutils"))]
+#[allow(dead_code)]
```

Comment:
why the `dead_code`? Should avoid adding this. If it's not used in tests, just gate it based on the feature `#[cfg(feature = "testutils")]` if it's caused when compiling the test target